cmake/Macros.cmake

```diff
@@ -127,7 +127,7 @@ function(AddHWGenTarget input template output)
 
   add_custom_command(OUTPUT ${output}
     COMMAND ${GEN_JSON} > ${output}
-    DEPENDS ${INPUT_JSON} ${LEGACY_JSON} ${TEMPLATE}
+    DEPENDS ${INPUT_JSON} ${LEGACY_JSON} ${TEMPLATE} ${RADIO_DIRECTORY}/src/hal/key_driver.h
```

Comment:
This should really not be here. Let me know if you have issues to fix this (CMake has its own ways, sometimes).

Reply:
If there's a better way to do this let me know - or just update the PR.

Reply:
How could the generation of a file from the JSON hardware definition depend on a that header file? Could the file generated be different, if the header changes? If this is not the case, I believe this dependence can actually be removed. Or what was the root problem at hand?

Reply:
The auto generated hal_heys.inc file contains the '_key_labels' array which is generated from the EnumKeys enum in key_driver.h. This PR removes an entry from the enum so the hal_keys.inc files need to be re-created. Without this you get a build error when switching branches between this one and main for example.

Reply:
Thx, I'll look into this.

Reply:
I see, it actually depends on `KEYS_LUT` defined in `generator.py`, and actually, the Python source should be what is actually defined in the dependencies. This is something that I did not do at the time, and probably should have.